Assets/Scripts/Buddy/BuddyStats.cs

```diff
@@ -283,56 +263,44 @@ void RestartEmoteRoutine()
 			StopCoroutine( _currentEmoteRoutine );
 		}
 
-		_currentEmoteRoutine = StartCoroutine( EmoteHappinessRoutine() );
+		_currentEmoteRoutine = StartCoroutine( EmoteHungerRoutine() );
 	}
 
-	IEnumerator EmoteHappinessRoutine()
+	IEnumerator EmoteHungerRoutine()
 	{
-		// Whether happiness or hunger was emoted last
-		bool didHappinessEmoteLast = MathUtils.RandBool();
-
 		while ( true )
 		{
 			yield return new WaitForSeconds( _emoteRoutineWait );
 
-			if ( didHappinessEmoteLast )
-			{
-				if ( _resources < _idealResourcesRange.min )
-				{
-					Emote( _hungryMaterial );
-				}
-				else if ( _resources > _idealResourcesRange.max )
-				{
-					Emote( _overFedMaterial );
-				}
-				else
-				{
-					Emote( _fullMaterial );
-				}
-
-				didHappinessEmoteLast = false;
-			}
-			else
-			{
-				if ( _happiness < _neutralHappinessRange.min )
-				{
-					Emote( _sadMaterial );
-				}
-				else if ( _happiness > _neutralHappinessRange.max )
-				{
-					Emote( _happyMaterial );
-				}
-				else
-				{
-					Emote( _neutralMaterial );
-				}
-
-				didHappinessEmoteLast = true;
-			}
+			UpdateHungerEmoteTexture();
+
+			EmoteHunger( _hungerEmoteMaterial );
+		}
+	}
+
+	private void UpdateHungerEmoteTexture()
+	{
+		float t;
+
+		if ( _resources < _idealResourcesRange.max )
+		{
+			t = (float) _resources / _idealResourcesRange.max;
+		}
+		else
+		{
+			t = 1f - (float) ( _resources - _idealResourcesRange.max ) / ( _idealResourcesRange.max - _idealResourcesRange.min );
```

Comment:
Rather than doing `_idealResourcesRange.max - _idealResourcesRange.min` you should be able to do `_idealResourceRange.range`. If that's not already a property you should add one.